File: Assignment_21/program3.c
// Accept N numbers from user and return the difference between largest and smallest number

///////////////////////////////////////////////////////////////////////////////
// File Name   :  program3.c
// Description :  Accept N numbers from user and return the difference between largest and smallest number
// Autor       :  Ankita Anil Patil
// Date        :  31/05/2025
///////////////////////////////////////////////////////////////////////////////


/*
 Input : N : 6
         Elements : 85 66 3 66 93 88
 Output: 90(93-3)

 */
  #include<stdio.h>
 #include<stdlib.h>

 int Difference(int Arr[], int iSize)
 {
    if((Arr == NULL) || (iSize <= 0))
    {
        printf("Invalid input");
        return -1;
    }

    int i = 0, iMax = 0, iMin = 0;

    iMax = Arr[0];
    iMin = Arr[0];

    for(i = 1; i < iSize; i++)
    {
        if(iMin > Arr[i])
        {
            iMin = Arr[i];
        }

        if(iMax < Arr[i])
        {
            iMax = Arr[i];
        }

    }
    return (iMax - iMin);

 }


 int main()
 {
    int iLength = 0, iCnt = 0, iRet = 0;
    int *ptr = NULL;

    printf("Enter number of elements \n");
    scanf("%d",&iLength);

    ptr = (int *) malloc(iLength * sizeof(int));

    if(NULL == ptr) 
    {
        printf("Unable to allocate memory");
        return -1;
    }

    printf("Enter elements\n");

    for(iCnt = 0; iCnt < iLength; iCnt++)
    {
        printf("Enter element %d :",iCnt+1);
        scanf("%d",&ptr[iCnt]);
    }

    iRet = Difference(ptr,iLength);
    printf("Difference between maximum and minimum is %d\n",iRet);

    free(ptr);

    return 0;
 }

File: Assignments_44/program3.java
// Write a program which accept matrix and return largest number from both the diagonals 
/* 
Input :

     3 2 5 9
     4 3 2 2
     8 4 1 5
     3 9 7 5

Output : 9
*/


///////////////////////////////////////////////////////////////////////////////
// File Name   :  program3.java
// Description :  Accept matrix and return largest number from both the diagonals 
// Autor       :  Ankita Anil Patil
// Date        :  2/08/2025
///////////////////////////////////////////////////////////////////////////////



import java.util.*;
class Matrix
{
    private  int Arr[][];

    public Matrix(int A, int B)
    {
        Arr = new int[A][B];
        
    }
    
    public void Accept()
    {
        System.out.println("Please enter elements of matrix ");
      
        Scanner sobj = new Scanner(System.in);

        int i = 0, j = 0;

        for(i = 0; i < Arr.length; i++)
        {
            for(j = 0 ; j < Arr[i].length; j++)
            {
                Arr[i][j] = sobj.nextInt();
            }
            
            System.out.println();
        }
     
    }


    public int MaxDiagonal() 
    { 
    
          int i = 0, j = 0, iMax1 = 0, iMax2 = 0;

          iMax1 = Arr[i][j];
          for(i = 0; i < Arr.length; i++)
          { 
               for(j = 0; j <Arr[i].length;j++)
               {
                    if(i == j)
                    {
                        if(iMax1 < Arr[i][j])
                        {
                         iMax1 = Arr[i][j];
                        }

                     }
                     
                     else if(j == (Arr.length -1) -i)
                     {
                        if(iMax2 < Arr[i][j])
                        {
                         iMax2 = Arr[i][j];
                        }

                     }
                    
               
               }
          }
   
        if(iMax1 > iMax2)
        {
          return iMax1;
        } 
        else
        {
          return iMax2;
        }  


   
    }

}


class program3
{
    public static void main(String A[]) 
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter Number of rows ");
        int iRow = sobj.nextInt();

        System.out.println("Enter Number of columns ");
        int iCol = sobj .nextInt();

        Matrix mobj = new Matrix(iRow,iCol);
        mobj.Accept();
        
        int iRet = 0;
        iRet = mobj.MaxDiagonal();

        System.out.println(" largest number from both the diagonals :" +iRet);

    }
}